To: vim_dev@googlegroups.com
Subject: Patch 7.3.362
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.362
Problem:    ml_get error when using ":g" with folded lines.
Solution:   Adjust the line number for changed_lines(). (Christian Brabandt)
Files:	    src/ex_cmds.c


*** ../vim-7.3.361/src/ex_cmds.c	2011-10-20 18:17:38.000000000 +0200
--- src/ex_cmds.c	2011-11-30 16:58:34.000000000 +0100
***************
*** 820,826 ****
  	curwin->w_cursor.lnum = dest + (line2 - line1) + 1;
  
      if (line1 < dest)
! 	changed_lines(line1, 0, dest + num_lines + 1, 0L);
      else
  	changed_lines(dest + 1, 0, line1 + num_lines, 0L);
  
--- 820,832 ----
  	curwin->w_cursor.lnum = dest + (line2 - line1) + 1;
  
      if (line1 < dest)
!     {
! 	dest += num_lines + 1;
! 	last_line = curbuf->b_ml.ml_line_count;
! 	if (dest > last_line + 1)
! 	    dest = last_line + 1;
! 	changed_lines(line1, 0, dest, 0L);
!     }
      else
  	changed_lines(dest + 1, 0, line1 + num_lines, 0L);
  
*** ../vim-7.3.361/src/version.c	2011-11-30 15:40:51.000000000 +0100
--- src/version.c	2011-11-30 17:01:07.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     362,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
218. Your spouse hands you a gift wrapped magnet with your PC's name
     on it and you accuse him or her of genocide.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
